client-cli/src/command/wallet_command.rs

```diff
@@ -5,27 +5,55 @@ use client_common::{Error, ErrorKind, Result};
 use client_core::WalletClient;
 
 use crate::ask_passphrase;
-
+use client_core::types::WalletKind;
 #[derive(Debug, StructOpt)]
 pub enum WalletCommand {
     #[structopt(name = "new", about = "New wallet")]
     New {
         #[structopt(name = "name", short, long, help = "Name of wallet")]
         name: String,
+        #[structopt(
+            name = "type",
+            short,
+            long,
+            help = "Type of wallet to create (hd, basic)"
+        )]
+        wallet_type: WalletKind,
     },
     #[structopt(name = "list", about = "List all wallets")]
     List,
+    #[structopt(name = "restore", about = "Restore HD Wallet")]
+    Restore {
+        #[structopt(name = "name", short, long, help = "Name of wallet")]
+        name: String,
+        #[structopt(
+            name = "mnemonic",
+            short,
+            long,
+            help = "Mnemonic of wallet (bip39 compatible mnemonics), such as \"word1 word2 ... \""
+        )]
+        mnemonic: String,
```

Comment:
perhaps better have it interactive -- if it is an argument, it'll end up logged in the shell history (and mnemonic ~ passphrase)

Reply:
ok, 
make sense
i'll change again